src/components/ProposalListTable.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {
+  Table,
+  TableHead,
+  ColumnHead,
+  TableBody,
+  TableRow,
+  RowCell,
+} from './Table';
+
+interface ProposalListTableRowProps {
+  // The `fields` array is currently required, as it determines the column order.
+  // If we can instead recompose the `proposal` children to be in the desired order,
+  // we don't need `fields`.
+  fields: {
+    shortCode: string,
+    label: string,
+  }[],
+  // Note the subtle difference from `proposals` on `ProposalListTableProps`:
+  // this is a single proposal, not an array of them.
+  proposal: {
+    fieldName: string,
+    value: string,
+  }[],
+  proposalId: string,
+}
+
+const ProposalListTableRow = ({
+  fields,
+  proposal,
+  proposalId,
+}: ProposalListTableRowProps) => {
+  const handleRowClick = () => {
+    // TODO: Replace this with a navigation to the proposal detail page.
+    // eslint-disable-next-line no-console
+    console.log(`Navigate to proposal detail page for ${proposalId}`);
+  };
+
+  return (
+    <TableRow onClick={handleRowClick}>
+      {/*
+        Iterate over fields to ensure order. Alternatively, if we can guarantee the proposal
+        field order matches the order of `fields`, we can do away with this and just iterate
+        over proposal fields sequentially.
+      */}
+      {fields.map((field) => {
+        // Note the two assumptions baked in here (also mentioned in the props design above):
+        // 1. Proposal `fieldName` corresponds to field `shortCode`
+        // 2. Each proposal has only one value for each matching field
+        const proposalField = proposal.find(({ fieldName }) => fieldName === field.shortCode);
+        return (
+          <RowCell key={field.shortCode}>
+            {proposalField?.value ?? ''}
+          </RowCell>
+        );
+      })}
+    </TableRow>
+  );
+};
+
+interface ProposalListTableProps {
+  /**
+   * An array of the fields that should be shown on this table.
+   * These correspond to "columns", but we're using a more semantic name.
```

Comment:
On the name specifically: you're right to accept "fields" from the page component, but I think we'll want to convert that to columns in this component.